mqsnoop/mqsnoop.bt

```diff
@@ -0,0 +1,34 @@
+#!/bin/env bpftrace
+// SPDX-License-Identifier: GPL-2.0-or-later
+/**
+ * mqsnoop.bt - Tracing and dump POSIX message queue send.
+ *              For Linux, uses bpftrace and eBPF.
+ *
+ * USAGE: mqsnoop.bt
+ *
+ * Copyright (C) 2024 CESTC, Co.
+ */
+BEGIN
+{
+	printf("Tracing POSIX message queue send, hit ctrl-c to end.\n");
+	printf("%-8s %-8s %-16s %-8s %-8s %-8s\n",
+		"TIME", "PID", "COMM", "LEN", "PRIO", "MSG");
+}
+
+/* No need to attach uprobe:/lib64/libc.so.6:mq_send, mq_send() will call
+ * mq_timedsend() */
+uprobe:/lib64/libc.so.6:mq_timedsend
```

Comment:
just curious what is the ".6" at the end of the shared library path?

Reply:
That's the current glibc major number :-)

Anyways, I'd suggest to use
```suggestion
uprobe:libc:mq_timedsend
```
as the `.so` files may be placed in different places in various distros.

Reply:
Fixed, thanks.